### PR TITLE
Rework: Reviews of sprint 2 and 3 

### PR DESCRIPTION
**Pull request by BarnabePILLIAUDIN**
# Done 

- All the review Avetis made

### CODE REVIEW

File: back/src/controllers/products/GetProductById.go [hunk 1]
```diff
@@ -10,12 +10,12 @@ import (
...
-	if err := SelectProductByID(&product, ctx); err != nil {
-		return helpers.SetStatusAndMessages(
-			constants.InternalServerErrorStatus,
-			constants.InternalServerErrorMessage,
-			ctx)
+	if err := SelectProductByID(&product, ctx); err == nil {
+		return sendProduct(ctx, product)
 	}
 
-	return sendProduct(ctx, product)
```
Comment (cedricgautier) [inline back/src/controllers/products/GetProductById.go]: Do not rework this, we need to keep sending the product.

**Review by cedricgautier — CHANGES_REQUESTED** [review]
One change is not needed, everything else looks good. Thanks

File: back/src/controllers/products/GetProductById.go (hunk 1 above)
Comment (barnabepilliaudin) [reply back/src/controllers/products/GetProductById.go]: We can still send products I just changed the condition but the behaviour of the code should be the same

Comment (cedricgautier) [reply back/src/controllers/products/GetProductById.go]: It creates a slight coding style change, I prefer that we don't change it if you agree.

Comment (barnabepilliaudin) [reply back/src/controllers/products/GetProductById.go]: Done

Comment (cedricgautier) [reply back/src/controllers/products/GetProductById.go]: Thanks

**Review by cedricgautier — APPROVED** [review]

**Review by KyDev25 — APPROVED** [review]
It's good for me!

**Pull request merged**